### PR TITLE
[CO-279] Document project architecture

### PR DESCRIPTION
**Pull request by HirotoShioi**
## Description

This PR introduces document regarding project architecture.
It tries to explain the following:
- General purpose of this project.
- How one can identify issue using log file.
- Use case diagram on how it works.
- Requirements
- Introduces its features.

## Linked issue

https://iohk.myjetbrains.com/youtrack/issue/CO-279

### CODE REVIEW

**Comment by HirotoShioi** [comment]
> We should switch to agent email / password authentication and not use an API token for this tool. That way we also ensure that only an agent can use this.

That is a good point, we can add that to the task.

> What do the circles represent? Haskell modules?

This is a use case diagram. (You can see more description regarding diagram [here](http://www.agilemodeling.com/artifacts/useCaseDiagram.htm))
Circle represents funtionality (action) log-classifier system provides to the Zendesk agent. I'll add the link to readme as well.

File: README.md [hunk 1]
```diff
@@ -1 +1,75 @@
...
+
+### Overview
+
+- Many of the Daedalus's issues can be identified by analyzing the log file. Classifier will utilize this by analyzing the log file and map with possible solution and problem which can be provided to the end user.
+- Ticket sent from Daedalus bug report has log file attached. Classifier will analyze the log file that is attached to the ticket. If any known issues where found, it'll then post an comment to that ticket. (See example below)
+- It uses [CSV](https://en.wikipedia.org/wiki/Comma-separated_values) (basically small record file) which contains all the information need to perform analysis. This make it so it is easy to add new known issues.
+- Zendesk agent will use [command line interface](https://en.wikipedia.org/wiki/Command-line_interface) in order to run the classifier.
+- Classifier will use the [Zendesk API](https://developer.zendesk.com/rest_api/docs/core/introduction) to perform any action. Because of this, agents will need to provide Zendesk token to the classifier.
```
Comment (StavrosChryselis) [inline README.md]: It's related to this line, let's change this to basic auth.

**Comment by HirotoShioi** [comment]
Fixed!

File: README.md [hunk 2]
```diff
@@ -1 +1,75 @@
...
+
+In order to use this, user must be IOHK Zendesk agent and be able to provide email and password to the classifier.
+
+## Features
+
+### Provide error text, problem, solution to the ticket
+
+Classifier will analyze the log file attached to the ticket. If any issues where found, it will post a comment which states
+
+- Errror code
```
[the remark this answers is not in the document]
Comment (HirotoShioi) [reply README.md]: @rokburgar Fixed! Thank you!

**Comment by HirotoShioi** [comment]
Added simple use-case scenario on when/how to use the classifier.

**Comment by ksaric** [comment]
@StavrosChryselis Please report this in a ticket and explain it a bit more - https://github.com/input-output-hk/log-classifier/pull/4#pullrequestreview-118453582

**Pull request merged**